### ISSUE TITLE
[BUG] Restoring a v1.25 RKE2/K3s cluster back to its original v1.24 K8s version with 1 etcd/cp and 3 worker nodes causes cluster to be stuck in `Updating`

### ISSUE THREAD

**Issue by markusewalker** [rancher/rancher#40843]
**Rancher Server Setup**
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/616b4b11719c1f8e0837b27b8fb97f300101406c
- Installation option (Docker install/Helm Chart): Docker
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: `v1.24`
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Linode

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Cluster Owner
  - If custom, define the set of permissions:

**Describe the bug**

As a standard user, I provisioned a downstream K3s v1.24 node driver cluster with the following configuration:
- 1 etcd/cp
- 3 worker nodes

Once done, I took an etcd snapshot and upgraded to K8s v1.25 with no issues. However, when I go to restore my snapshot using all restore options, my cluster is stuck in the following `Updating` state toggling between these two:

![image](https://user-images.githubusercontent.com/52591819/224190565-e8d2738e-be43-42ff-9052-bd1479aa42fc.png)

![image](https://user-images.githubusercontent.com/52591819/224190577-03c018c0-4580-43be-a7b3-12f4e114d4cb.png)

Looking into the Rancher logs, I see the following:

```
2023/03/09 23:37:08 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: waiting for etcd restore bootstrap restart
2023/03/09 23:37:12 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: waiting for etcd restore worker restart
2023/03/09 23:37:12 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: waiting for etcd restore worker restart
2023/03/09 23:37:15 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: refreshing etcd restore state
2023/03/09 23:37:15 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:15 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:15 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already
2023/03/09 23:37:15 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already
2023/03/09 23:37:18 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:19 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for probes: kube-apiserver, kube-controller-manager, kube-scheduler, kubelet
2023/03/09 23:37:19 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for probes: kube-apiserver, kube-controller-manager, kube-scheduler, kubelet
2023/03/09 23:37:19 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already
2023/03/09 23:37:21 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for probes: kube-apiserver, kube-controller-manager, kube-scheduler, kubelet
2023/03/09 23:37:25 [ERROR] [CAPI] Reconciler error: failed to add &Node{ObjectMeta:{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] [] []},Spec:NodeSpec{PodCIDR:,DoNotUseExternalID:,ProviderID:,Unschedulable:false,Taints:[]Taint{},ConfigSource:nil,PodCIDRs:[],},Status:NodeStatus{Capacity:ResourceList{},Allocatable:ResourceList{},Phase:,Conditions:[]NodeCondition{},Addresses:[]NodeAddress{},DaemonEndpoints:NodeDaemonEndpoints{KubeletEndpoint:DaemonEndpoint{Port:0,},},NodeInfo:NodeSystemInfo{MachineID:,SystemUUID:,BootID:,KernelVersion:,OSImage:,ContainerRuntimeVersion:,KubeletVersion:,KubeProxyVersion:,OperatingSystem:,Architecture:,},Images:[]ContainerImage{},VolumesInUse:[],VolumesAttached:[]AttachedVolume{},Config:nil,},} watch on cluster fleet-default/mew-k3s: failed to create cluster accessor: error creating dynamic rest mapper for remote cluster "fleet-default/mew-k3s": Get "https://10.43.248.221/k8s/clusters/c-m-9zv9sgw5/api?timeout=10s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
2023/03/09 23:37:26 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for probes: kube-apiserver, kube-controller-manager, kube-scheduler, kubelet
2023/03/09 23:37:30 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already
2023/03/09 23:37:31 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for probes: kube-apiserver, kube-controller-manager, kube-scheduler, kubelet
2023/03/09 23:37:33 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:34 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:34 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already
2023/03/09 23:37:35 [ERROR] [CAPI] Reconciler error: failed to add &Node{ObjectMeta:{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] [] []},Spec:NodeSpec{PodCIDR:,DoNotUseExternalID:,ProviderID:,Unschedulable:false,Taints:[]Taint{},ConfigSource:nil,PodCIDRs:[],},Status:NodeStatus{Capacity:ResourceList{},Allocatable:ResourceList{},Phase:,Conditions:[]NodeCondition{},Addresses:[]NodeAddress{},DaemonEndpoints:NodeDaemonEndpoints{KubeletEndpoint:DaemonEndpoint{Port:0,},},NodeInfo:NodeSystemInfo{MachineID:,SystemUUID:,BootID:,KernelVersion:,OSImage:,ContainerRuntimeVersion:,KubeletVersion:,KubeProxyVersion:,OperatingSystem:,Architecture:,},Images:[]ContainerImage{},VolumesInUse:[],VolumesAttached:[]AttachedVolume{},Config:nil,},} watch on cluster fleet-default/mew-k3s: failed to create cluster accessor: error creating dynamic rest mapper for remote cluster "fleet-default/mew-k3s": Get "https://10.43.248.221/k8s/clusters/c-m-9zv9sgw5/api?timeout=10s": context deadline exceeded
2023/03/09 23:37:39 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:43 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:44 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:45 [ERROR] [CAPI] Reconciler error: failed to add &Node{ObjectMeta:{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] [] []},Spec:NodeSpec{PodCIDR:,DoNotUseExternalID:,ProviderID:,Unschedulable:false,Taints:[]Taint{},ConfigSource:nil,PodCIDRs:[],},Status:NodeStatus{Capacity:ResourceList{},Allocatable:ResourceList{},Phase:,Conditions:[]NodeCondition{},Addresses:[]NodeAddress{},DaemonEndpoints:NodeDaemonEndpoints{KubeletEndpoint:DaemonEndpoint{Port:0,},},NodeInfo:NodeSystemInfo{MachineID:,SystemUUID:,BootID:,KernelVersion:,OSImage:,ContainerRuntimeVersion:,KubeletVersion:,KubeProxyVersion:,OperatingSystem:,Architecture:,},Images:[]ContainerImage{},VolumesInUse:[],VolumesAttached:[]AttachedVolume{},Config:nil,},} watch on cluster fleet-default/mew-k3s: failed to create cluster accessor: error creating dynamic rest mapper for remote cluster "fleet-default/mew-k3s": Get "https://10.43.248.221/k8s/clusters/c-m-9zv9sgw5/api?timeout=10s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
2023/03/09 23:37:45 [ERROR] [CAPI] Unable to retrieve Node status: failed to create cluster accessor: failed to get lock for cluster: cluster is locked already
2023/03/09 23:37:49 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:53 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for cluster agent to connect
2023/03/09 23:37:54 [INFO] [planner] rkecluster fleet-default/mew-k3s: waiting: configuring bootstrap node(s) mew-k3s-etcd-cp-8c8788f5-zv6pg: waiting for probes: kube-apiserver, kube-controller-manager, kube-scheduler, kubelet
```
This behavior seems to be specific to v1.24 -> v1.25 as when I go from v1.23 -> v1.24 or even v1.23 -> v1.25, I do not witness the above issue.

**To Reproduce**

1. As a standard user, provision a downstream K3s v1.24 node driver cluster.
2. Validate that the cluster and nodes come up in an `Active` state in Cluster Management and Cluster Explorer.
3. Take an etcd snapshot of the cluster in this state.
4. Edit the cluster configuration and upgrade K8s version to v1.25.
5. Validate that the cluster and nodes go into `Updating` and eventually back `Active` with the upgraded K8s version in Cluster Management and Cluster Explorer.
6. Restore the cluster to v1.24 using restore options cluster configuration, etcd and Kubernetes version.
7. Note that the cluster is stuck in `Updating` and never restores back to v1.24.

**Result**
The cluster is stuck in `Updating` state and toggles back and forth with message `waiting for cluster agent to connect` and `waiting for probes: kube-apiserver, kube-controller-manager, kube-scheduler, kubelet`.

**Expected Result**
The cluster should successfully restore back to v1.24 with no issues.

**Comment by jakefhyde** [comment]
The k3s server is crashlooping on the downstream node, and the following log message can be seen:

```
Mar 14 18:02:29 jhyde-k3s-test-server-9d74a8c8-cf5xb k3s[797207]: time="2023-03-14T18:02:29Z" level=info msg="Creating service-controller event broadcaster"
Mar 14 18:02:29 jhyde-k3s-test-server-9d74a8c8-cf5xb k3s[797207]: W0314 18:02:29.242337  797207 dispatcher.go:153] Failed calling webhook, failing closed rancher.cattle.io.namespaces: failed calling webhook "rancher.cattle.io.namespaces": failed to call webhook: Post "[https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation/namespaces?timeout=10s](https://rancher-webhook.cattle-system.svc/v1/webhook/validation/namespaces?timeout=10s)": proxy error from 127.0.0.1:6443 while dialing 10.42.1.3:9443, code 503: 503 Service Unavailable
Mar 14 18:02:29 jhyde-k3s-test-server-9d74a8c8-cf5xb k3s[797207]: time="2023-03-14T18:02:29Z" level=fatal msg="Failed to register service-controller handlers: Internal error occurred: failed calling webhook \"rancher.cattle.io.namespaces\": failed to call webhook: Post \"[https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation/namespaces?timeout=10s](https://rancher-webhook.cattle-system.svc/v1/webhook/validation/namespaces?timeout=10s)\": proxy error from 127.0.0.1:6443 while dialing 10.42.1.3:9443, code 503: 503 Service Unavailable"
Mar 14 18:02:29 jhyde-k3s-test-server-9d74a8c8-cf5xb systemd[1]: k3s.service: Main process exited, code=exited, status=1/FAILURE
Mar 14 18:02:29 jhyde-k3s-test-server-9d74a8c8-cf5xb systemd[1]: k3s.service: Failed with result 'exit-code'.
```

This is identical to the log message [here](https://github.com/k3s-io/k3s/issues/6815) which a fix has been merged for as of `v1.24.11+k3s1` which is now available in rancher. I confirmed that this issue no longer happens when using `v1.24.11+k3s1`, taking a snapshot, upgrading to `v1.25.7+k3s1`, and then performing a restore. RKE2 downstream clusters also have a similar, but different issue, but the cluster agent eventually connects and there is no crashlooping, and the intended fixes have not yet hit RKE2.

**Comment by sowmyav27** [comment]
Reopening since there is an active discussion happening with this issue. cc: @snasovich @jakefhyde to provide info on the actual issue and any potential fix/workarounds needed.

**Comment by jakefhyde** [comment]
The underlying issue is the same as https://github.com/k3s-io/k3s/issues/6815, although the intended fix (which has already been merged in k3s) does not address previous k3s versions. As a result, further discussion was required across teams regarding a potential workaround. After some consideration, a workaround has been accepted by both teams and will be implemented by team 1 to address this issue. This comment will be updated once the details of the workaround are finalized.

**Comment by KevinJoiner** [comment]
# Validation

## Root Cause
K3s would try to create the `kube-system` namespace to verify it exists but would get denied because the rancher-webhook was not yet running. This would cause a fatal error causing K3s to exit and reboot. On reboot, the same scenario would occur, causing a crash loop of the cluster. 

## What was fixed, or what change has occurred
We updated K3s to first look for the existence of the namespace before attempting to create it, causing it to no-longer crash. 
This fix only allows computability with v2.7.2 with newer versions of K3s.

We made an additional change to rancher-webhook to allow backwards compatibility with previous versions of K3s that do not have the above fix.
This Change enables requests to succeed when the webhook is down if and only if the request is to Create the `kube-system` namespace and the rancher-webhook is unreachable. This does not bypass all request validation on namespace creation, only specific Rancher validation.

## Areas or cases that should be tested
The restarting of K3s with one of the two fixes missing.
Since the error is a race condition between k3s and the rancher-webhook pod, we scale down the webhook deployment to reliably reproduce the fault scenario.

### validate k3s fix
1. As a standard user, provision a downstream K3s v1.24.11 > node driver cluster.
2. Validate that the cluster and nodes come up in an Active state in Cluster Management and Cluster Explorer.
3. Open a shell to the downstream cluster.
4. Edit the rancher-webhook image to `rancher-webhook:v0.3.2-rc21`  so that the webhook fix is not included 
   - `kubectl edit deployment rancher-webhook -n cattle-system`
5. Scale down rancher-webhook to 0 replicas `kubectl scale deploy rancher-webhook -n cattle-system --replicas=0`
6. Wait 20s for the new rancher-webhook to start and update the validatingAdmissionVonfiguration.
7. Restart K3s on the cluster with `systemctl restart k3s` or Perform an upgrade and restore to v1.25
8. Validate that the cluster and nodes go into Updating and eventually back Active.

### validate webhook fix
1. As a standard user, provision a downstream K3s v1.24 node driver cluster.
2. Validate that the cluster and nodes come up in an Active state in Cluster Management and Cluster Explorer.
3. Open a shell to the local cluster.
4. Edit the k3s version to `v1.24.10` so that the k3s fix is not included  
   - `kubectl edit clusters.provisioning.cattle.io <downstream_name> -n fleet-default -o yaml`
5. Scale down rancher-webhook to 0 replicas `kubectl scale deploy rancher-webhook -n cattle-system --replicas=0`
6. Restart K3s on the cluster with `systemctl restart k3s` or Perform an upgrade and restore to v1.25
7. Validate that the cluster and nodes go into Updating and eventually back Active.



## What areas could experience regressions
Validation of PSA labels

- Create a rancher on v2.7-head
- Provision a downstream cluster 
- Open a shell to the downstream cluster
- Verify that the admin can create and edit namespaces and arbitrary namespace 'foo'
- Stop rancher-webhook `kubectl scale deploy rancher-webhook -n cattle-system --replicas=0`
- Verify that attempts to create and edit namespaces in the downstream cluster fail.
  
## Are the repro steps accurate/minimal?
Since there are two fixes to this problem, we need to revert our downstream not to include either of the changes

### v1.23 -> v1.25 -> v1.23
1. As a standard user, provision a downstream K3s v1.24.11 > node driver cluster.
2. Validate that the cluster and nodes come up in an Active state in Cluster Management and Cluster Explorer.
3. Open a shell to the local cluster.
4. Edit the k3s version to `v1.24.10` so that the k3s fix is not included  
   - `kubectl edit clusters.provisioning.cattle.io <downstream_name> -n fleet-default -o yaml`
3. Open a shell to the downstream cluster.
4. Edit the rancher-webhook image to `rancher-webhook:v0.3.2-rc21`  so that the webhook fix is not included 
   - `kubectl edit deployment rancher-webhook -n cattle-system`
5. Scale down rancher-webhook to 0 replicas `kubectl scale deploy rancher-webhook -n cattle-system --replicas=0`
6. Wait 20s for the new rancher-webhook to start and update the validatingAdmissionVonfiguration.
7. Restart K3s on the cluster with `systemctl restart k3s` or Perform an upgrade and restore to v1.25
8. Validate that the cluster and nodes go into Updating and never recover.

**Comment by markusewalker** [comment]
**QA TEST PLAN**

Scenarios
|Scenario|Test Case|
|---|---|
|1|As a standard user, provision a downstream K3s v1.24.11 node driver cluster, create an etcd snapshot, upgrade to v1.25, restore the snapshot, validate the cluster reverts to v1.24.11|
|2|As a standard user, provision a downstream K3s v1.24.11 node driver cluster, edit the rancher-webhook image, scale the rancher-webhook to 0 replicas, create an etcd snapshot, upgrade to v1.25, restore the snapshot|
|3|As a standard user, provision a downstream K3s v1.24.10 custom cluster, scale rancher-webhook to 0 replicas, create an etcd snapshot, restore to v1.25, restore the snapshot|
|4|Setup Rancher v2.7.1, provision a downstream K3s v1.24.10 custom cluster, upgrade Rancher v2.7.1 -> v2.7-head, upgrade K8s version to v1.24.11, edit the rancher-webhook image, scale the rancher-webhook to 0 replicas, create an etcd snapshot, upgrade to v1.25, restore the snapshot|
|5|Setup Rancher v2.7.1, provision a downstream K3s v1.24.10 custom cluster, upgrade Rancher v2.7.1 -> v2.7-head, scale rancher-webhook to 0 replicas, create an etcd snapshot, restore to v1.25, restore the snapshot|

**Comment by markusewalker** [comment]
Validated that this is addressed in `v2.7-head` https://github.com/rancher/rancher/commit/a841a9e101ae601a033fa8b39342ad4800c91d85.

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/a841a9e101ae601a033fa8b39342ad4800c91d85
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, provision a downstream K3s v1.24.11 node driver cluster, create an etcd snapshot, upgrade to v1.25, restore the snapshot, validate the cluster reverts to v1.24.11|:white_check_mark:|
|2|As a standard user, provision a downstream K3s v1.24.11 node driver cluster, edit the rancher-webhook image, scale the rancher-webhook to 0 replicas, create an etcd snapshot, upgrade to v1.25, restore the snapshot|:white_check_mark:|
|3|As a standard user, provision a downstream K3s v1.24.10 custom cluster, scale rancher-webhook to 0 replicas, create an etcd snapshot, restore to v1.25, restore the snapshot|:white_check_mark:|
|4|Setup Rancher v2.7.1, provision a downstream K3s v1.24.10 node driver cluster, upgrade Rancher v2.7.1 -> v2.7-head, upgrade K8s version to v1.24.11, edit the rancher-webhook image, scale the rancher-webhook to 0 replicas, create an etcd snapshot, upgrade to v1.25, restore the snapshot|:white_check_mark:|
|5|Setup Rancher v2.7.1, provision a downstream K3s v1.24.10 node driver cluster, upgrade Rancher v2.7.1 -> v2.7-head, scale rancher-webhook to 0 replicas, create an etcd snapshot, restore to v1.25, restore the snapshot|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. As a standard user, navigated to Cluster Management > Create.
2. Provisioned a downstream K3s v1.24.11 node driver cluster (1 etcd/cp, 3 workers) with no PSACT set.
3. Validated that the cluster and nodes came up as `Active` in Cluster Management and Cluster Explorer.
4. Validated that all pods in the system project page came up as `Active`.
5. Created an etcd snapshot of the cluster as is.
6. Edited the cluster's configuration to upgrade the K8s version to v1.25 and PSACT set to `rancher-restricted`.
7. Validated that the cluster and nodes upgraded their K8s versions in Cluster Management and Cluster Explorer and is `Active`.
8. Created a test deployment in the `default` namespace and validated that this fails due to my PSACT being `rancher-restricted`.
9. Restored the snapshot with all options; validated the cluster, cluster configuration and nodes are back at v1.24.11 and are `Active`.
10. Validated that the created deployment no longer appears after restoring the snapshot.
______
**Scenario 2**
1. As a standard user, navigated to Cluster Management > Create.
2. Provisioned a downstream K3s v1.24.11 node driver cluster (1 etcd/cp, 3 workers) with no PSACT set.
3. Validated that the cluster and nodes came up as `Active` in Cluster Management and Cluster Explorer.
4. Validated that all pods in the system project page came up as `Active`.
5. Navigated to top-left menu > Workloads > Deployments.
6. Edited the `rancher-webhook` deployment to have image `rancher-webhook:0.3.2-rc21`.
7. Still in the `rancher-webhook` deployment, updated the `Replicas` to be 0.
8. Validated that the new `rancher-webhook` is `Active` with this change.
9. Created an etcd snapshot. 
10. Edited the cluster's configuration to upgrade the K8s version to v1.25 and PSACT set to `rancher-restricted`.
11. Validated that the cluster and nodes upgraded their K8s versions in Cluster Management and Cluster Explorer and is `Active`.
12. Created a test deployment in the `default` namespace and validated that this fails due to my PSACT being `rancher-restricted`.
13. Restored the snapshot with all options; validated the cluster, cluster configuration and nodes are back at v1.24.11 and are `Active`.
14. Validated that the created deployment no longer appears after restoring the snapshot.
______
**Scenario 3**
1. As a standard user, navigated to Cluster Management > Create > Custom.
2. Provisioned a downstream K3s v1.24.10 custom cluster (1 etcd/cp, 3 workers).
3. Registered the applicable registration command to each node in the cluster.
4. Validated that the cluster and nodes came up as `Active` in Cluster Management and Cluster Explorer.
5. Validated that all pods in the system project page came up as `Active`.
6. Navigated to top-left menu > Workloads > Deployments.
7. Scaled down the `rancher-webhook` replicas from 1 to 0 and saved the change.
8. Validated that the new `rancher-webhook` is `Active` with this change.
9. Created an etcd snapshot. 
10. Edited the cluster's configuration to upgrade the K8s version to v1.25 and PSACT set to `rancher-restricted`.
11. Validated that the cluster and nodes upgraded their K8s versions in Cluster Management and Cluster Explorer and is `Active`.
12. Created a test deployment in the `default` namespace and validated that this fails due to my PSACT being `rancher-restricted`.
13. Restored the snapshot with all options; validated the cluster, cluster configuration and nodes are back at v1.24.11 and are `Active`.
14. Validated that the created deployment no longer appears after restoring the snapshot.
_____
**Scenario 4**
1. Repeated scenario 2, steps 1-5.
2. Upgraded Rancher v2.7.1 -> v2.7-head; validated that the cluster did not auto-upgrade and that the cluster and nodes remained in a state of `Active`.
3. Edited the cluster's configuration and updated the K8s version to v1.24.11.
4. Validated that the cluster and nodes upgraded successfully in Cluster Management and Cluster Explorer.
5. Repeated scenario 2, steps 6-14 with the same, successful results.
________
**Scenario 5**
1. Repeated scenario 3, steps 1-5.
2. Upgraded Rancher v2.7.1 -> v2.7-head; validated that the cluster did not auto-upgrade and that the cluster and nodes remained in a state of `Active`.
3. Repeated scenario 3, steps 6-14 with the same, successful results.
